src/test/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTripTest.java

```diff
@@ -214,6 +220,65 @@ private static Stream<Arguments> createDelayNotificationTestCases() {
         );
     }
 
+    @ParameterizedTest
+    @MethodSource("createLegTransitionNotificationTestCases")
+    void testLegTransitionNotifications(
+        NotificationType legTransitionType,
+        String travelerName,
+        TravelerPosition travelerPosition,
+        Locale locale,
+        String message
+    ) {
+        TripMonitorNotification[] notification = LegTransitionNotification.createLegTransitionNotifications(
+            List.of(legTransitionType),
+            travelerName,
+            travelerPosition,
+            locale
+        );
+        assertNotNull(notification[0]);
+        assertEquals(message, notification[0].body);
+    }
+
+    private static Stream<Arguments> createLegTransitionNotificationTestCases() throws Exception {
+        String travelerName = "Obi-Wan";
+        Locale locale = Locale.US;
+        Itinerary itinerary = createDefaultItinerary();
+        Leg expectedLeg = itinerary.legs.get(1);
+        Coordinates expectedLegDestinationCoords = new Coordinates(expectedLeg.to);
+        Leg nextLeg = itinerary.legs.get(2);
+        Coordinates nextLegDepartureCoords = new Coordinates(nextLeg.from);
+        return Stream.of(
+            Arguments.of(
+                NotificationType.MODE_CHANGE_NOTIFICATION,
+                travelerName,
+                new TravelerPosition(expectedLeg, nextLeg, expectedLegDestinationCoords),
+                locale,
+                "Obi-Wan has changed transit from TRAM to WALK."
```

Comment:
Refine the text to say that "[traveler] has arrived at transit stop [stop name]", and omit the specific modes.

Reply:
Updated in: https://github.com/ibi-group/otp-middleware/pull/275/commits/91d66190cb611283993b8e745148641efd695c5e